aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,360 @@
+		configuration := raw.(map[string]interface{})
+		m := globalaccelerator.EndpointConfiguration{}
+
+		m.EndpointId = aws.String(configuration["endpoint_id"].(string))
+		m.Weight = aws.Int64(int64(configuration["weight"].(int)))
+
+		out[i] = &m
+	}
+
+	log.Printf("[DEBUG] Expand endpoint_configurations: %s", out)
+	return out
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupFlattenEndpointDescriptions(configurations []*globalaccelerator.EndpointDescription) []interface{} {
+	out := make([]interface{}, len(configurations))
+
+	for i, configuration := range configurations {
+		m := make(map[string]interface{})
+
+		m["endpoint_id"] = *configuration.EndpointId
+		m["weight"] = *configuration.Weight
+
+		out[i] = m
+	}
+
+	log.Printf("[DEBUG] Flatten endpoint_configurations: %s", out)
+	return out
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupRetrieve(conn *globalaccelerator.GlobalAccelerator, endpointGroupArn string) (*globalaccelerator.EndpointGroup, error) {
+	resp, err := conn.DescribeEndpointGroup(&globalaccelerator.DescribeEndpointGroupInput{
+		EndpointGroupArn: aws.String(endpointGroupArn),
+	})
+
+	if err != nil {
+		if isAWSErr(err, globalaccelerator.ErrCodeEndpointGroupNotFoundException, "") {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return resp.EndpointGroup, nil
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).globalacceleratorconn
+
+	opts := &globalaccelerator.UpdateEndpointGroupInput{
+		EndpointGroupArn: aws.String(d.Id()),
+	}
+
+	if v, ok := d.GetOk("health_check_interval_seconds"); ok {
+		opts.HealthCheckIntervalSeconds = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_path"); ok {
+		opts.HealthCheckPath = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("health_check_port"); ok {
+		opts.HealthCheckPort = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_protocol"); ok {
+		opts.HealthCheckProtocol = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("threshold_count"); ok {
+		opts.ThresholdCount = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("traffic_dial_percentage"); ok {
+		opts.TrafficDialPercentage = aws.Float64(v.(float64))
+	}
+
+	if v, ok := d.GetOk("endpoint_configurations"); ok {
+		opts.EndpointConfigurations = resourceAwsGlobalAcceleratorEndpointGroupExpandEndpointConfigurations(v.([]interface{}))
+	} else {
+		opts.EndpointConfigurations = []*globalaccelerator.EndpointConfiguration{}
+	}
+
+	log.Printf("[DEBUG] Update Global Accelerator endpoint group: %s", opts)
+
+	_, err := conn.UpdateEndpointGroup(opts)
+	if err != nil {
+		return fmt.Errorf("Error updating Global Accelerator endpoint group: %s", err)
+	}
+
+	acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Creating an endpoint group triggers the accelerator to change status to InPending
```

Comment:
This logic should reuse the available function (albeit slightly poor naming): `resourceAwsGlobalAcceleratorAcceleratorWaitForState`

```go
err = resourceAwsGlobalAcceleratorAcceleratorWaitForState(conn, acceleratorArn)
if err != nil {
	return err
}
```